Date: Fri Feb 19 11:46:54 GMT 2021
From: Fri Feb 19 11:46:54 GMT 2021
Subject: AMLOGIC-1069 Option to adjust HDMI IN audio delay
Source: SKY
License: CLOSED
Upstream-Status: Pending
Signed-off-by: Suraj Nadiger <suraj.nadiger@sky.uk>

Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -19,6 +19,7 @@ TEST_HAL_PARAM_OBJS=src/hal_param.o
 TEST_HAL_DUMP_OBJS=src/hal_dump.o
 TEST_HAL_PATCH_OBJS=src/hal_patch.o
 TEST_MASTER_VOL_OBJS=src/master_vol.o
+TEST_HDMI_DELAY_OBJS=src/hdmi_delay.o
 EFFECT_TOOL_OBJS=src/effect_tool.o
 
 PROTOC=$(HOST_DIR)/bin/protoc
@@ -50,7 +51,7 @@ src/audio_server.cpp: src/audio_service.
 src/audio_client.cpp: src/audio_service.pb.h src/audio_service.grpc.pb.cc
 src/audio_if_client.cpp: src/audio_service.pb.h src/audio_service.grpc.pb.cc
 
-all: audio_server libaudio_client.so audio_client_test audio_client_test_ac3 halplay dap_setting speaker_delay digital_mode test_arc start_arc hal_param hal_dump hal_patch master_vol effect_tool
+all: audio_server libaudio_client.so audio_client_test audio_client_test_ac3 halplay dap_setting speaker_delay digital_mode test_arc start_arc hal_param hal_dump hal_patch master_vol effect_tool hdmi_delay
 
 audio_server: $(SERVER_OBJS)
 	$(CC) $(CFLAGS) $(SC_LDFLAGS) -o $@ $^
@@ -97,6 +98,9 @@ master_vol: $(TEST_MASTER_VOL_OBJS) liba
 effect_tool: $(EFFECT_TOOL_OBJS) libaudio_client.so
 	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
+hdmi_delay: $(TEST_HDMI_DELAY_OBJS) libaudio_client.so
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
+
 .PHONY: install
 install:
 	install -m 755 -D audio_server -t $(TARGET_DIR)/usr/bin/
@@ -113,6 +117,7 @@ install:
 	install -m 755 -D hal_patch $(TARGET_DIR)/usr/bin/
 	install -m 755 -D master_vol $(TARGET_DIR)/usr/bin/
 	install -m 755 -D effect_tool $(TARGET_DIR)/usr/bin/
+	install -m 755 -D hdmi_delay $(TARGET_DIR)/usr/bin/
 	install -m 644 -D libaudio_client.so -t $(TARGET_DIR)/usr/lib/
 	install -m 644 -D libaudio_client.so -t $(STAGING_DIR)/usr/lib/
 	install -m 644 -D include/audio_if_client.h -t $(STAGING_DIR)/usr/include
@@ -139,6 +144,7 @@ clean:
 	rm -f hal_patch
 	rm -f master_vol
 	rm -f effect_tool
+	rm -f hdmi_delay
 	rm -rf $(STAGING_DIR)/usr/include/hardware
 	rm -rf $(STAGING_DIR)/usr/include/system
 	rm -f libaudio_client.so
@@ -155,6 +161,7 @@ clean:
 	rm -f $(TARGET_DIR)/usr/bin/hal_patch
 	rm -f $(TARGET_DIR)/usr/bin/master_vol
 	rm -f $(TARGET_DIR)/usr/bin/effect_tool
+	rm -f $(TARGET_DIR)/usr/bin/hdmi_delay
 	rm -f $(TARGET_DIR)/usr/lib/libaudio_client.so
 	rm -f $(STAGING_DIR)/usr/lib/libaudio_client.so
 	rm -f $(STAGING_DIR)/usr/include/audio_if_client.h
Index: git/src/hdmi_delay.c
===================================================================
--- /dev/null
+++ git/src/hdmi_delay.c
@@ -0,0 +1,44 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <audio_if.h>
+
+#define HDMI_IN_DELAY_CMD "delay_input_hdmi_ms="
+
+int main(int argc, char **argv)
+{
+    audio_hw_device_t *device;
+    int delay = 0, ret;
+    char cmd[256];
+
+    if (argc < 2) {
+        printf("Expected delay_input_hdmi_ms <delay in ms>\n");
+        return -1;
+    }
+
+    delay = atoi(argv[1]);
+    if ((delay < 0) || (delay >= 250)) {
+        printf("Delay out of range\n");
+        return -2;
+    }
+    snprintf(cmd, sizeof(cmd), "%s%d", HDMI_IN_DELAY_CMD, delay);
+
+    ret = audio_hw_load_interface(&device);
+    if (ret) {
+        fprintf(stderr, "audio_hw_load_interface failed: %d\n", ret);
+        return ret;
+    }
+
+    ret = device->init_check(device);
+    if (ret) {
+        printf("device not initiated, quit\n");
+        audio_hw_unload_interface(device);
+        return -1;
+    }
+
+    ret = device->set_parameters(device, cmd);
+    printf("set_parameters returns %d\n", ret);
+
+    audio_hw_unload_interface(device);
+
+    return ret;
+}
