diff --git a/package/nodejs/Config.in b/package/nodejs/Config.in
index 1489306be0..b5aaaa167b 100644
--- a/package/nodejs/Config.in
+++ b/package/nodejs/Config.in
@@ -44,6 +44,13 @@ config BR2_PACKAGE_NODEJS
 
 if BR2_PACKAGE_NODEJS
 
+config BR2_PACKAGE_NODEJS_LIBRARY
+	bool "library"
+        default n
+	help
+          Build the NodeJS packag as a library and an executable 
+          using the library.
+
 config BR2_PACKAGE_NODEJS_NPM
 	bool "NPM for the target"
 	select BR2_PACKAGE_OPENSSL
diff --git a/package/nodejs/nodejs.mk b/package/nodejs/nodejs.mk
index 367d5d2058..1e3945c234 100644
--- a/package/nodejs/nodejs.mk
+++ b/package/nodejs/nodejs.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-NODEJS_VERSION = 16.18.1
+NODEJS_VERSION = 18.16.0
 NODEJS_SOURCE = node-v$(NODEJS_VERSION).tar.xz
 NODEJS_SITE = http://nodejs.org/dist/v$(NODEJS_VERSION)
 NODEJS_DEPENDENCIES = \
@@ -41,6 +41,10 @@ NODEJS_CONF_OPTS = \
 	--dest-os=linux \
 	--ninja
 
+ifeq ($(BR2_PACKAGE_NODEJS_LIBRARY),y)
+NODEJS_CONF_OPTS += --shared
+endif
+
 HOST_NODEJS_MAKE_OPTS = \
 	$(HOST_CONFIGURE_OPTS) \
 	CXXFLAGS="$(HOST_NODEJS_CXXFLAGS)" \
diff --git a/package/nodejs/0005-shared-lib-installation.patch b/package/nodejs/0005-shared-lib-installation.patch
new file mode 100644
index 0000000000..dab60f748f
--- /dev/null
+++ b/package/nodejs/0005-shared-lib-installation.patch
@@ -0,0 +1,10 @@
+--- a/tools/install.py	2022-11-04 12:22:08.000000000 +0100
++++ b/tools/install.py	2023-04-19 00:41:42.093186113 +0200
+@@ -150,6 +150,7 @@
+   action([output_prefix + output_file], 'bin/' + output_file)
+ 
+   if 'true' == variables.get('node_shared'):
++    output_prefix = output_prefix + 'lib/'
+     if is_windows:
+       action([output_prefix + 'libnode.dll'], 'bin/libnode.dll')
+       action([output_prefix + 'libnode.lib'], 'lib/libnode.lib')
diff --git a/package/nodejs/0001-add-qemu-wrapper-support.patch b/package/nodejs/0001-add-qemu-wrapper-support.patch
index 184f3d19b7..c316a33d6e 100644
--- a/package/nodejs/0001-add-qemu-wrapper-support.patch
+++ b/package/nodejs/0001-add-qemu-wrapper-support.patch
@@ -17,32 +17,9 @@ host-builds.
 
 Signed-off-by: Adam Duskett <aduskett@gmail.com>
 ---
- node.gyp                 |  4 ++--
  tools/v8_gypfiles/v8.gyp | 10 +++++-----
- 2 files changed, 7 insertions(+), 7 deletions(-)
+ 1 file changed, 7 insertions(+), 7 deletions(-)
 
-diff --git a/node.gyp b/node.gyp
-index 97cdbb531d..db5cf51ce6 100644
---- a/node.gyp
-+++ b/node.gyp
-@@ -319,7 +319,7 @@
-               'action_name': 'run_mkcodecache',
-               'process_outputs_as_sources': 1,
-               'inputs': [
--                '<(mkcodecache_exec)',
-+                @MAYBE_WRAPPER@ '<(mkcodecache_exec)',
-               ],
-               'outputs': [
-                 '<(SHARED_INTERMEDIATE_DIR)/node_code_cache.cc',
-@@ -346,7 +346,7 @@
-                   'action_name': 'node_mksnapshot',
-                   'process_outputs_as_sources': 1,
-                   'inputs': [
--                    '<(node_mksnapshot_exec)',
-+                    @MAYBE_WRAPPER@ '<(node_mksnapshot_exec)',
-                     '<(node_snapshot_main)',
-                   ],
-                   'outputs': [
 diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
 index 73099a03e4..88ca91efb4 100644
 --- a/tools/v8_gypfiles/v8.gyp
diff --git a/package/nodejs/nodejs.hash b/package/nodejs/nodejs.hash
index 4408782248..ef71cc4873 100644
--- a/package/nodejs/nodejs.hash
+++ b/package/nodejs/nodejs.hash
@@ -1,5 +1,6 @@
 # From https://nodejs.org/dist/v16.18.1/SHASUMS256.txt
 sha256  1f8051a88f86f42064f4415fe7a980e59b0a502ecc8def583f6303bc4d445238  node-v16.18.1.tar.xz
+sha256	33d81a233e235a509adda4a4f2209008d04591979de6b3f0f67c1c906093f118  node-v18.16.0.tar.xz
 
 # Hash for license file
 sha256  0bec08634ba79b5404f6b7f92ea850f3c2a06e27e6f83f2267e4f5e55ae33334  LICENSE
--- /dev/null	2023-05-30 13:09:48.302074425 +0200
+++ 0006-std-library-usage-fix.patch	2023-06-27 14:27:00.998756710 +0200
@@ -0,0 +1,11 @@
+--- a/deps/v8/include/v8-internal.h	2023-04-12 06:09:14.000000000 +0200
++++ b/deps/v8/include/v8-internal.h	2023-06-27 14:07:32.594066538 +0200
+@@ -643,7 +643,7 @@
+ template <class T>
+ V8_INLINE void PerformCastCheck(T* data) {
+   CastCheck<std::is_base_of<Data, T>::value &&
+-            !std::is_same<Data, std::remove_cv_t<T>>::value>::Perform(data);
++            !std::is_same<Data, std::remove_cv<T>>::value>::Perform(data);
+ }
+ 
+ // A base class for backing stores, which is needed due to vagaries of
