diff --git a/package/nodejs/Config.in b/package/nodejs/Config.in
index 1489306be0..b5aaaa167b 100644
--- a/package/nodejs/Config.in
+++ b/package/nodejs/Config.in
@@ -44,6 +44,13 @@ config BR2_PACKAGE_NODEJS
 
 if BR2_PACKAGE_NODEJS
 
+config BR2_PACKAGE_NODEJS_LIBRARY
+	bool "library"
+        default n
+	help
+          Build the NodeJS packag as a library and an executable 
+          using the library.
+
 config BR2_PACKAGE_NODEJS_NPM
 	bool "NPM for the target"
 	select BR2_PACKAGE_OPENSSL
diff --git a/package/nodejs/nodejs-src/nodejs-src.mk b/package/nodejs/nodejs-src/nodejs-src.mk
index 8f934202b0..de7750868d 100644
--- a/package/nodejs/nodejs-src/nodejs-src.mk
+++ b/package/nodejs/nodejs-src/nodejs-src.mk
@@ -70,6 +70,10 @@ endef
 HOST_NODEJS_SRC_PRE_CONFIGURE_HOOKS += NODEJS_SRC_PYTHON3_SYMLINK
 NODEJS_SRC_PRE_CONFIGURE_HOOKS += NODEJS_SRC_PYTHON3_SYMLINK
 
+ifeq ($(BR2_PACKAGE_NODEJS_LIBRARY),y)
+NODEJS_SRC_CONF_OPTS += --shared
+endif
+
 ifeq ($(BR2_PACKAGE_OPENSSL),y)
 NODEJS_SRC_DEPENDENCIES += openssl
 NODEJS_SRC_CONF_OPTS += --shared-openssl

--- /dev/null	2025-06-04 05:10:23.124001072 +0000
+++ a/package/nodejs/nodejs-src/0005-library-install-path-fix.patch	2025-06-04 06:53:41.681683485 +0000
@@ -0,0 +1,11 @@
+--- a/tools/install.org	2025-06-04 06:39:18.850331689 +0000
++++ b/tools/install.py	2025-06-04 06:41:49.562504930 +0000
+@@ -181,7 +181,7 @@
+       link_path = abspath(options.install_path, 'lib/libnode.so')
+       try_symlink(options, so_name, link_path)
+     else:
+-      output_lib = 'libnode.' + options.variables.get('shlib_suffix')
++      output_lib = 'lib/libnode.' + options.variables.get('shlib_suffix')
+       action(options, [os.path.join(options.build_dir, output_lib)],
+              os.path.join(options.variables.get('libdir'), output_lib))
+ 
