diff --git a/package/nodejs/Config.in b/package/nodejs/Config.in
index 1489306be0..b5aaaa167b 100644
--- a/package/nodejs/Config.in
+++ b/package/nodejs/Config.in
@@ -44,6 +44,13 @@ config BR2_PACKAGE_NODEJS
 
 if BR2_PACKAGE_NODEJS
 
+config BR2_PACKAGE_NODEJS_LIBRARY
+	bool "library"
+        default n
+	help
+          Build the NodeJS packag as a library and an executable 
+          using the library.
+
 config BR2_PACKAGE_NODEJS_NPM
 	bool "NPM for the target"
 	select BR2_PACKAGE_OPENSSL
diff --git a/package/nodejs/nodejs-src/0005-shared-lib-installation.patch b/package/nodejs/nodejs-src/0005-shared-lib-installation.patch
new file mode 100644
index 0000000000..dab60f748f
--- /dev/null
+++ b/package/nodejs/nodejs-src/0005-shared-lib-installation.patch
@@ -0,0 +1,10 @@
+--- a/tools/install.py	2022-11-04 12:22:08.000000000 +0100
++++ b/tools/install.py	2023-04-19 00:41:42.093186113 +0200
+@@ -150,6 +150,7 @@
+   action([output_prefix + output_file], 'bin/' + output_file)
+ 
+   if 'true' == variables.get('node_shared'):
++    output_prefix = output_prefix + 'lib/'
+     if is_windows:
+       action([output_prefix + 'libnode.dll'], 'bin/libnode.dll')
+       action([output_prefix + 'libnode.lib'], 'lib/libnode.lib')
diff --git a/package/nodejs/nodejs-src/nodejs-src.mk b/package/nodejs/nodejs-src/nodejs-src.mk
index 8f934202b0..de7750868d 100644
--- a/package/nodejs/nodejs-src/nodejs-src.mk
+++ b/package/nodejs/nodejs-src/nodejs-src.mk
@@ -70,6 +70,10 @@ endef
 HOST_NODEJS_SRC_PRE_CONFIGURE_HOOKS += NODEJS_SRC_PYTHON3_SYMLINK
 NODEJS_SRC_PRE_CONFIGURE_HOOKS += NODEJS_SRC_PYTHON3_SYMLINK
 
+ifeq ($(BR2_PACKAGE_NODEJS_LIBRARY),y)
+NODEJS_SRC_CONF_OPTS += --shared
+endif
+
 ifeq ($(BR2_PACKAGE_OPENSSL),y)
 NODEJS_SRC_DEPENDENCIES += openssl
 NODEJS_SRC_CONF_OPTS += --shared-openssl

