diff --git a/package/nodejs/Config.in b/package/nodejs/Config.in
index 1489306be0..b5aaaa167b 100644
--- a/package/nodejs/Config.in
+++ b/package/nodejs/Config.in
@@ -44,6 +44,13 @@ config BR2_PACKAGE_NODEJS
 
 if BR2_PACKAGE_NODEJS
 
+config BR2_PACKAGE_NODEJS_LIBRARY
+	bool "library"
+        default n
+	help
+          Build the NodeJS packag as a library and an executable 
+          using the library.
+
 config BR2_PACKAGE_NODEJS_NPM
 	bool "NPM for the target"
 	select BR2_PACKAGE_OPENSSL
diff --git a/package/nodejs/nodejs-src/nodejs-src.mk b/package/nodejs/nodejs-src/nodejs-src.mk
index 8f934202b0..de7750868d 100644
--- a/package/nodejs/nodejs-src/nodejs-src.mk
+++ b/package/nodejs/nodejs-src/nodejs-src.mk
@@ -70,6 +70,10 @@ endef
 HOST_NODEJS_SRC_PRE_CONFIGURE_HOOKS += NODEJS_SRC_PYTHON3_SYMLINK
 NODEJS_SRC_PRE_CONFIGURE_HOOKS += NODEJS_SRC_PYTHON3_SYMLINK
 
+ifeq ($(BR2_PACKAGE_NODEJS_LIBRARY),y)
+NODEJS_SRC_CONF_OPTS += --shared
+endif
+
 ifeq ($(BR2_PACKAGE_OPENSSL),y)
 NODEJS_SRC_DEPENDENCIES += openssl
 NODEJS_SRC_CONF_OPTS += --shared-openssl

--- /dev/null
+++ b/package/nodejs/nodejs-src/0006-add-cstdarg-include-for-strings.patch
@@ -0,0 +1,11 @@
+--- a/deps/v8/src/base/strings.org	2025-03-29 07:58:55.749254386 +0000
++++ b/deps/v8/src/base/strings.h	2025-03-29 07:59:10.969711062 +0000
+@@ -9,6 +9,8 @@
+ #include "src/base/macros.h"
+ #include "src/base/vector.h"
+ 
++#include <cstdarg>
++
+ namespace v8 {
+ namespace base {
+
--- /dev/null	2024-09-23 10:38:10.659163567 +0200
+++ b/package/nodejs/nodejs-src/0005-shared-lib-installation.patch	2024-09-23 11:08:49.867138620 +0200
@@ -0,0 +1,11 @@
+--- a/tools/install.org	2024-09-23 11:07:15.974139893 +0200
++++ b/tools/install.py		2024-09-23 11:07:27.607139736 +0200
+@@ -175,7 +175,7 @@
+       link_path = abspath(options.install_path, 'lib/libnode.so')
+       try_symlink(options, so_name, link_path)
+     else:
+-      output_lib = 'libnode.' + options.variables.get('shlib_suffix')
++      output_lib = 'lib/libnode.' + options.variables.get('shlib_suffix')
+       action(options, [os.path.join(options.build_dir, output_lib)],
+              os.path.join(options.variables.get('libdir'), output_lib))
+  
