diff --git a/Source/WPEFramework/scripts/wpeframework.in b/Source/WPEFramework/scripts/wpeframework.in
index f96973d7..ccd9d12f 100755
--- a/Source/WPEFramework/scripts/wpeframework.in
+++ b/Source/WPEFramework/scripts/wpeframework.in
@@ -17,6 +17,11 @@ start() {
     # needed for wayland/westeros/weston
     export XDG_RUNTIME_DIR=/tmp
 
+    # preload westeros gl stuff
+	if [ -e /usr/lib/libwesteros_gl.so ] && [ -e /usr/share/WPEFramework/Compositor/libcompositorplugin.so ]; then
+        export LD_PRELOAD=libwesteros_gl.so
+	fi
+
     echo -n "Starting WPEFramework: "
     start-stop-daemon -S -q -b -m -p /var/run/WPEFramework.pid --exec /usr/bin/WPEFramework -- -b </dev/null >/dev/null 2>&1
     [ $? == 0 ] && echo "OK" || echo "FAIL"
