diff --git a/linux/Config.in b/linux/Config.in
index 2f37a2e418..25ed7f93d4 100644
--- a/linux/Config.in
+++ b/linux/Config.in
@@ -222,6 +222,14 @@ config BR2_LINUX_KERNEL_CUSTOM_LOGO_PATH
 	  converted to the linux kernel format (224 colors only)
 	  and copied over the original logo file.
 
+config BR2_LINUX_COMMON_DRIVERS
+	string "External drivers"
+	help
+	  If there are prvate drivers but that are required during the kernel
+          build, place them in the linx source tree and indicate the path to
+          the drivers here, so they can be included in the config and in the 
+          build
+
 #
 # Binary format
 #
diff --git a/linux/linux.mk b/linux/linux.mk
index 12699b8ee4..ca899648e1 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -70,6 +70,7 @@ LINUX_PATCH = $(filter ftp://% http://% https://%,$(LINUX_PATCHES))
 # host-pkgconf will look for host libraries and not target ones.
 LINUX_MAKE_ENV = \
 	$(HOST_MAKE_ENV) \
+	COMMON_DRIVERS_DIR=$(call qstrip,$(BR2_LINUX_COMMON_DRIVERS)) \
 	BR_BINARIES_DIR=$(BINARIES_DIR)
 
 LINUX_INSTALL_IMAGES = YES
