--- a/arch/arm/boot/dts/allwinner/sun8i-h3.dtsi	2025-10-10 14:51:36.263831755 +0200
+++ b/arch/arm/boot/dts/allwinner/sun8i-h3.dtsi	2025-10-10 14:28:02.579563901 +0200
@@ -44,6 +44,12 @@
 #include <dt-bindings/thermal/thermal.h>
 
 / {
+
+	aliases {
+		mmc0 = &mmc0;    /* SD card slot */
+		mmc1 = &mmc1;    /* Wi-Fi / eMMC / secondary interface */
+	};
+
 	cpu0_opp_table: opp-table-cpu {
 		compatible = "operating-points-v2";
 		opp-shared;
